query/src/main/java/org/infinispan/query/impl/QueryDefinition.java

```diff
@@ -98,26 +85,20 @@ private QueryEngine getQueryEngine(AdvancedCache<?, ?> cache) {
 
    public void initialize(AdvancedCache<?, ?> cache) {
       if (hsQuery == null) {
-         if (luceneQuery != null) {
-            hsQuery = ComponentRegistryUtils.getSearchIntegrator(cache).createHSQuery(luceneQuery);
-            if (sort != null)
-               hsQuery.sort(sort);
+         QueryEngine<?> queryEngine = getQueryEngine(cache);
+         HsQueryRequest hsQueryRequest;
+         if (indexedType != null && sortableFields != null) {
+            IndexedTypeMap<CustomTypeMetadata> metadata = createMetadata();
+            hsQueryRequest = queryEngine.createHsQuery(queryString, metadata, namedParameters);
          } else {
-            QueryEngine queryEngine = getQueryEngine(cache);
-            HsQueryRequest hsQueryRequest;
-            if (indexedType != null && sortableFields != null) {
-               IndexedTypeMap<CustomTypeMetadata> metadata = createMetadata();
-               hsQueryRequest = queryEngine.createHsQuery(queryString, metadata, namedParameters);
-            } else {
-               hsQueryRequest = queryEngine.createHsQuery(queryString, null, namedParameters);
-            }
-            hsQuery = hsQueryRequest.getHsQuery();
-            sort = hsQueryRequest.getSort();
-            hsQuery.projection(hsQueryRequest.getProjections());
+            hsQueryRequest = queryEngine.createHsQuery(queryString, null, namedParameters);
          }
-         hsQuery.firstResult(firstResult);
-         hsQuery.maxResults(maxResults);
+         hsQuery = hsQueryRequest.getHsQuery();
+         sort = hsQueryRequest.getSort();
+         hsQuery.projection(hsQueryRequest.getProjections());
       }
+      hsQuery.firstResult(firstResult);
```

Comment:
The old version of the code was setting firstResult and maxResults here only first time, if hsQuery == null.